lib/trento/application/integration/discovery/policies/sap_system_policy.ex

```diff
@@ -46,25 +46,30 @@ defmodule Trento.Integration.Discovery.SapSystemPolicy do
          },
          host_id
        ) do
+    # credo:disable-for-lines:10 Credo.Check.Refactor.Nesting
     Enum.flat_map(databases, fn %{"Database" => tenant} ->
       Enum.map(
         instances,
-        fn {_, instance} ->
-          instance_number = parse_instance_number(instance)
-
-          RegisterDatabaseInstance.new(%{
-            sap_system_id: UUID.uuid5(@uuid_namespace, id),
-            sid: sid,
-            tenant: tenant,
-            host_id: host_id,
-            instance_number: instance_number,
-            instance_hostname: parse_instance_hostname(instance, instance_number),
-            features: parse_features(instance, instance_number),
-            http_port: parse_http_port(instance, instance_number),
-            https_port: parse_https_port(instance, instance_number),
-            start_priority: parse_start_priority(instance, instance_number),
-            health: parse_instance_health(instance, instance_number)
-          })
+        fn instance ->
+          case parse_instance_number(instance) do
+            {:ok, instance_number} ->
+              RegisterDatabaseInstance.new(%{
+                sap_system_id: UUID.uuid5(@uuid_namespace, id),
+                sid: sid,
+                tenant: tenant,
+                host_id: host_id,
+                instance_number: instance_number,
+                instance_hostname: parse_instance_hostname(instance, instance_number),
+                features: parse_features(instance, instance_number),
+                http_port: parse_http_port(instance, instance_number),
+                https_port: parse_https_port(instance, instance_number),
+                start_priority: parse_start_priority(instance, instance_number),
+                health: parse_instance_health(instance, instance_number)
+              })
+
+            {:error, reason} ->
+              {:error, reason}
+          end
         end
       )
```

Comment:
The newline for the `)` looks a bit weird, is this the formatter? (it's not doing the same below for some reason :/)

Reply:
Changed! `mix format` behaves a bit different depending on how you format the function arguments